plugins/async-node/core/src/index.test.ts

```diff
@@ -547,3 +547,43 @@ test("should call onAsyncNode hook when async node is encountered", async () =>
     expect(localNode.type).toStrictEqual("async");
   });
 });
+
+test("missing node-id", async () => {
```

Comment:
not sure i'm following this test, the node in here doesn't actually get parsed to be an async node, does it?

Reply:
Here I'm testing for an invalid async node (one that has the `async` property but not the `id` property) that is a child of another property within the view. Here I'm specifically testing that the async node plugin returns a `null` AST by hitting L168 in `plugins/async-node/core/src/index.ts`.

Reply:
I see. The thing I have to point out here is that even if it's a valid async node, the _view_ would still be what you are checking because unresolved async nodes don't actually add anything to the view tree.

I'd check to see if the hook ever gets called for processing async nodes, or something along those lines

Reply:
Simply returning `null` on L168 in `plugins/async-node/core/src/index.ts` instead of doing the `childOptions` check, results in a view tree that looks like this `{ id: 'my-view', fields: { async: 'true' } }`. So there is a faulty async node still added to the tree, unless I'm missing something here.

Reply:
try adding an id to that nested object, I think you'll likely still get the test passing which would mean id or no id, the view is the same

Reply:
Ahhh ok! I see what you mean.